scripts/dashboard/validate_dashboards_format.py

```diff
@@ -0,0 +1,168 @@
+'''This script will be triggered by a github action outlined in ~/.github/workflows/validate_dashboards_formar.yml.
+It will be used to validate all new json and metadata files pertaining to integration sample dashboards.'''
+
+import json
+import os
+import sys
+import yaml
+
+class MetadataFormattingError(Exception):
+  '''Raised when data in metadata file does not have expected format'''
+  pass
+
+class JsonFormattingError(Exception):
+  '''Raised when data in json file or file name does not have expected format'''
+  pass
+
+class MetadataInJsonMisMatchError(Exception):
+  '''Raised when data in metadata file does not have matching data in a json file'''
+  pass
+
+class JsonInMetadataMisMatchError(Exception):
+  '''Raised when data in json file does not have matching data in a metadata file'''
+  pass
+
+def validate_json(dashboards_data):
+  try:
+    json.dumps(dashboards_data)
+  except:
+    return False
+
+  return True
+
+def map_paths(paths):
+  '''Takes in list of paths and returns dictionary of metadata and dashboard json data, 
+  organized by directories'''
+  map = {}
+
+  for path in paths:
+    path_parts = path.split('/')
+
+    # filter out non-dashboards files
+    if not 'dashboards' == path_parts[1] or len(path_parts) > 4:
+      continue
+
+    if not path_parts[2] in map:
+      map[path_parts[2]] = {
+        'metadata': {
+        },
+        'json_data': {
+        },
+        'json_files': []
+      }
+
+    if path_parts[3].split('.')[-1] == "json":
+      map[path_parts[2]]['json_files'].append(path_parts[3])
+
+  return map
+
+def get_sample_dashboards_json(path):
+  '''Checks if file is in json format and returns data in dictionary form'''
+  with open(path) as f:
+    dashboards_dict = json.load(f)
+
+    if not validate_json(dashboards_dict):
```

Comment:
I think json.load(f) is what's going to fail and throw here. If .load() is successful, would .dumps() fail?

Reply:
yeah I think you're right. Does that mean json.dumps was useless in the alerts checks I originally based this script off of? It seemed like the only reason they used json.dumps was for a check.

I can move the json.load method to the validation function.

Reply:
done.